src/main/java/edu/cornell/kfs/pmw/batch/businessobject/service/impl/PaymentWorksVendorSearchService.java

```diff
@@ -0,0 +1,107 @@
+package edu.cornell.kfs.pmw.batch.businessobject.service.impl;
+
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.tuple.Pair;
+import org.kuali.kfs.core.api.search.SearchOperator;
+import org.kuali.kfs.krad.bo.BusinessObjectBase;
+import org.kuali.kfs.krad.util.ObjectUtils;
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.kfs.sys.businessobject.service.impl.DefaultSearchService;
+import org.springframework.util.MultiValueMap;
+
+import edu.cornell.kfs.pmw.batch.PaymentWorksPropertiesConstants.PaymentWorksVendor;
+
+public class PaymentWorksVendorSearchService extends DefaultSearchService {
+
+    @Override
+    protected MultiValueMap<String, String> transformSearchParams(
+            final Class<? extends BusinessObjectBase> boClass,
+            final MultiValueMap<String, String> searchParams) {
+        final MultiValueMap<String, String> transformedSearchParams =
+                super.transformSearchParams(boClass, searchParams);
+
+        transformVendorNumberParams(transformedSearchParams);
+        transformProcessTimestampParams(transformedSearchParams);
+
+        return transformedSearchParams;
+    }
+
+    private void transformVendorNumberParams(final MultiValueMap<String, String> transformedSearchParams) {
+        if (transformedSearchParams.containsKey(PaymentWorksVendor.KFS_VENDOR_NUMBER)) {
+            final String kfsVendorNumber = transformedSearchParams
+                    .remove(PaymentWorksVendor.KFS_VENDOR_NUMBER)
+                    .get(0);
+            final Pair<String, String> parsedKfsVendorNumber = parseKfsVendorNumber(kfsVendorNumber);
+            if (StringUtils.isNotBlank(parsedKfsVendorNumber.getLeft())) {
+                transformedSearchParams.put(
+                        PaymentWorksVendor.KFS_VENDOR_HEADER_GENERATED_IDENTIFIER,
+                        List.of(parsedKfsVendorNumber.getLeft()));
+            }
+            if (StringUtils.isNotBlank(parsedKfsVendorNumber.getRight())) {
+                transformedSearchParams.put(
+                        PaymentWorksVendor.KFS_VENDOR_DETAIL_ASSIGNED_IDENTIFIER,
+                        List.of(parsedKfsVendorNumber.getRight()));
+            }
+        }
+    }
+
+    private Pair<String, String> parseKfsVendorNumber(final String kfsVendorNumber) {
+        if (StringUtils.isBlank(kfsVendorNumber)) {
+            return Pair.of(KFSConstants.EMPTY_STRING, KFSConstants.EMPTY_STRING);
+        } else if (!StringUtils.contains(kfsVendorNumber, KFSConstants.DASH)) {
+            return Pair.of(kfsVendorNumber, KFSConstants.EMPTY_STRING);
+        }
+        final String vendorHeaderId = StringUtils.substringBefore(kfsVendorNumber, KFSConstants.DASH);
```

Comment:
I had configured the lookup to support string-based vendor number searching so that users could specify wildcards if necessary. If the vendor number string is malformed or contains too many hyphens, then the string-based search should result in zero matches.

If you think it would be better to have such malformed strings result in search criteria errors instead, please let me know.

Reply:
Thanks for the explanation, leaving things as they are is fine.